### ISSUE TITLE
Implement different network architectures (resnet, deepnet) with batch norm

### ISSUE THREAD

**Issue by raspstephan** [raspstephan/CBRAIN-Keras-Diagnostics#7]
This is closely connected with #1.

I recently listened to this podcast with Jeremy Howard: https://twimlai.com/twiml-talk-214-trends-in-deep-learning-with-jeremy-howard/

In it he talks about how people realized that batch norm and skip connections change the loss surface, making networks easier to train: https://arxiv.org/abs/1712.09913

I wonder if part of the stability problems we are experiencing are due to a very spiky minimum during training with bad generalization capabilities. 

I tried batch norm offline at some point but it didn't increase the generalization to "new" climates offline. But maybe this is a separate issue.
